flow/Arena.h

```diff
@@ -208,26 +209,32 @@ inline void save( Archive& ar, const Arena& p ) {
 	// No action required
 }
 
+// Optional is a wrapper for std::optional. There
+// are two primary reasons to use this wrapper instead
+// of using std::optional directly:
+//
+// 1) Legacy: A lot of code was written using Optional before
+//    std::optional was available.
+// 2) When you call get but no value is present Optional gives an
+//    assertion failure. std::optional, on the other hand, would
+//    throw std::bad_optional_access.
```

Comment:
Why is asserting better than throwing?

I'm guessing it's because we don't want people to attempt to catch the exception, but maybe it's just because it would be too hard to inspect all the call sites.

(Update the comment)

Reply:
We don't tend to handle exceptions that aren't our own exceptions particularly well. For example, in many contexts a non-flow exception will be converted to an `unknown_error`, and tracking that down is a lot more difficult than an assert that gives you a backtrace where things went wrong.

Reply:
Totally agree.  I'm also not a fan of exceptions that shouldn't be caught.